### PR TITLE
Disallow ampdoc in Services.actionServiceForDoc()

### PR DESCRIPTION
**Pull request by choumx**
Partial for #16322.

#### Approach
Only allow passing `!Element|!ShadowRoot` to service getters for current FIE-supported services (actions, bind, url, urlReplacements).

#### Rationale 
Though not all extensions modified in this PR are supported in A4A (e.g. amp-story), this has the nice benefit that they will automatically do the right thing in FIE context if A4A decides to support them. `amp-gwd-animation` recently underwent this transition, for example.

By contrast, if we decided to continue using `AmpDoc` for FIE, we'd need to worry if callers are passing the correct `AmpDoc` instance.

#### Follow-ups
- [ ] Do the same thing for bind, url, and urlReplacements.
- [ ] Rename `actionServiceForDoc` to `actionForElement`.

/cc @dvoytenko

### CODE REVIEW

File: extensions/amp-access/0.1/amp-access.js [hunk 1]
```diff
@@ -293,9 +293,8 @@ export class AccessService {
 
   /** @private */
   startInternal_() {
-    // TODO(dvoytenko, #3742): This will refer to the ampdoc once AccessService
-    // is migrated to ampdoc as well.
-    Services.actionServiceForDoc(this.ampdoc).installActionHandler(
+    const actionService = Services.actionServiceForDoc(this.accessElement_);
```
Comment (dvoytenko) [inline extensions/amp-access/0.1/amp-access.js]: Does this actually work in shadow mode? I just realized that `accessElement` is actually very unusual - it's not in the doc's `body`, but in `head`. There could be some differences in this regard.

File: src/service.js [hunk 2]
```diff
@@ -111,22 +111,31 @@ export function getExistingServiceInEmbedScope(win, id, opt_fallbackToTopWin) {
...
+  const local = getLocalExistingServiceForEmbedWinOrNull(win, id);
+  if (local) {
+    return local;
   }
-  // If an ampdoc is passed or fallback is allowed, continue resolving.
-  if (!nodeOrDoc.nodeType || opt_fallbackToTopWin) {
-    return getServiceForDocDeprecated(nodeOrDoc, id);
+  // Next, node is embedded AND fallback is not allowed, return null.
+  // Otherwise, resolve via ampdoc.
+  const isEmbed = win != getTopWindow(win);
```
Comment (dvoytenko) [inline src/service.js]: If the section right above is about "local embed window", why not then just check `const isEmbed = ...; if (isEmbed...)` before calling the `getLocalExistingServiceForEmbedWinOrNull` above?

File: src/service/standard-actions-impl.js [hunk 3]
```diff
@@ -54,7 +54,7 @@ export class StandardActions {
     this.ampdoc = ampdoc;
 
     /** @const @private {!./action-impl.ActionService} */
-    this.actions_ = Services.actionServiceForDoc(ampdoc);
+    this.actions_ = Services.actionServiceForDoc(ampdoc.getHeadNode());
```
Comment (dvoytenko) [inline src/service/standard-actions-impl.js]: Ditto: I have some concerns about the head node for shadow docs.

**Comment by alanorozco** [comment]
Changes to `src/service/video-manager-impl.js` LGTM

**Comment by newmuis** [comment]
Story changes LGTM

File: extensions/amp-access/0.1/amp-access.js (hunk 1 above)
Comment (dvoytenko) [reply extensions/amp-access/0.1/amp-access.js]: `node.ownerDocument` is not an active path in the shadow mode however.

Comment (choumx) [reply extensions/amp-access/0.1/amp-access.js]: Sorry, got mixed up -- I meant that the FIE path is the same. 

For PWA, it follows the `AmpDocService.getAmpDoc()` code path which walks up the shadow root: https://github.com/ampproject/amphtml/blob/master/src/service/ampdoc-impl.js#L180

**Comment by choumx** [comment]
Thanks Dima!

**Pull request merged**